examples/ts/proxy-usage.ts

```diff
@@ -0,0 +1,41 @@
+
+import ccxt from '../../js/ccxt.js';
+
+// // AUTO-TRANSPILE //
+
+async function example_proxyUrl () {
+    const myEx = new ccxt.kucoin ();
+    myEx.proxyUrl = 'http://5.75.153.75:8090/proxy.php?url=';
+    console.log (await myEx.fetch ('https://api.ipify.org/'));
+}
+
+async function example_httpProxy () {
+    const myEx = new ccxt.kucoin ();
+    myEx.httpProxy = 'http://5.75.153.75:8888'; // "httpProxy" or "httpsProxy" (depending on your proxy protocol)
+    console.log (await myEx.fetch ('https://api.ipify.org/'));
+}
+
+async function example_socksProxy () {
+    const myEx = new ccxt.kucoin ();
+    myEx.socksProxy = 'socks5://127.0.0.1:1080'; // socks5 or socks5h
+    console.log (await myEx.fetch ('https://api.ipify.org/'));
+}
+
+async function example_webSockets () {
+    const myEx = new ccxt.pro.kucoin ();
+    await myEx.initializeProxies (); // only needed for WEBSOCKETS in JAVASCRIPT, not in PYTHON or PHP
```

Comment:
@ttodua I'm a bit confused by this requirement,
any exchange will start by loading the markets through REST right? so `initializeProxies` will always be called first inside fetch, no?

Reply:
it's requirement only if user wants to use proxies. if user does not want to use proxies, we should not load/import them. not? 
that's why I've left that as explicit call/requirement for users who want to use proxies. should I amend it?

Reply:
What's the difference between this call, after instantiating and this one inside fetch? 
![image](https://github.com/ccxt/ccxt/assets/43336371/6aa6059d-e190-4332-bf82-d317b438c0b5)

Reply:
oha oha, seems you'r right, we are already loading it inside fetch. my assumption went wrong bcz i was solely using that example which you referenced above (which didn't load market and didn't make any call at all, thus there was needed to explicitly call). 
but as you pointed out perfectly, it does in fetch already

Reply:
so, specifically in "example" ts file, I'll change that comment, thus we neither need to call it explicitly in JS (unless user is trying to do same thing as in that example file:)

Reply:
however, that means we are still loading proxy modules for all users (even for them that don't use them). lmk wdyt? maybe it's not critical, bcz that's done only once per lifecycle.

Reply:
though we still load module in memory, and if users are loading 100+ exchanges, then they might be wasting some minor RAM (idk how much RAM though, might be insignificant..).

Reply:
the problem is that, if we don't load the proxies as mandatory in firstmost `fetch` call (on loadMarkets), then if user loads exchange markets without proxy, and then sets proxy after it:
```
await e.loadMarkets();
....
e.httpProxy = ..;
await e.watchX();
```
then the "fetch" call might not happen in between the proxy setting and `watchX` method, so there would be needed to call `await e.,initializeProxies()` in between `e.httpProxy` and `e.watchX()` lines. 

so, loading proxy modules in first fetch seems somewhat "workaround" atm (even though the above mentioned disadvantage of loading it for all users).

let me know your decision.

Reply:
@ttodua in that case we can have a method called `refreshProxies` or something (might be an alias for initializeProxies) and we can document that if we suddenly change the proxies, they need to be refreshed

Reply:
I have two thoughts:
- `refreshProxies` might be a bit vague name. can we name it something more related, like "recheckProxyModules" or alike, which would better express what does the method do?

Reply:
- so, it will be only for JS.. I am still asking myself, whether there could be some better way to avoid users having different functionality (which entails reading manuals and etc) so users didn't need that explicit "refreshProxies" call ever... it is possible, that we exported module from ccxt base, and it was not dynamically imported later? so, the `http(s)Proxy` module could get instantiated only in case if user uses proxy settings. in this way: (A) all users would have loaded proxy module only once (even thought they loaded hundreds of exchanges) but the proxy instance will not be added into instance property for those 100 exchanges, but only for those, where user sets proxies.